Testing/VelaptorTests/OpenGL/GLWindowTests.cs

```diff
@@ -872,45 +1259,177 @@ public void GLWindow_WhenRenderingFrame_InvokesDrawAndSwapsBuffer()
+                drawInvoked = true;
+            };
+            window.AutoClearBuffer = true;
             window.Show();
 
             // Act
-            this.mockWindowFacade.Raise(m => m.Unload += null, EventArgs.Empty);
+            this.mockSilkWindow.Raise(e => e.Closing += null);
+            this.mockSilkWindow.Raise(e => e.Render += It.IsAny<Action<double>>(), 0.016);
 
             // Assert
+            this.mockGL.VerifyNever(m => m.Clear(It.IsAny<GLClearBufferMask>()));
+            Assert.False(drawInvoked, $"The '{nameof(GLWindow.Draw)}()' method should not of been invoked.");
+            this.mockGLContext.VerifyNever(m => m.SwapBuffers());
+        }
+
+        [Fact]
+        public void GLWindow_WhenUnloadingWindow_ShutsDownWindow()
+        {
+            // Arrange
+            var uninitializeInvoked = false;
+            var window = CreateWindow();
+            window.Uninitialize += () => uninitializeInvoked = true;
+            window.Show();
+
+            // Act
+            this.mockSilkWindow.Raise(e => e.Closing += null);
+
+            // Assert
+            Assert.True(uninitializeInvoked);
             this.mockShutDownReactable.Verify(m => m.PushNotification(default, true), Times.Once);
             this.mockShutDownReactable.Verify(m => m.Dispose(), Times.Once);
         }
 
+        [Fact]
+        public void GLWindow_WhenKeyboardKeyIsPressedDown_UpdatesKeyboardInputState()
+        {
+            // Arrange
+            var window = CreateWindow();
+            window.Show();
+
+            // Act
+            this.mockSilkKeyboard.Raise(e => e.KeyDown += It.IsAny<Action<IKeyboard, Key, int>>(),
+                null,
+                Key.Space,
+                0);
+
+            // Assert
+            this.mockKeyInput.VerifyOnce(m => m.SetState(KeyCode.Space, true));
+        }
+
+        [Fact]
+        public void GLWindow_WhenKeyboardKeyIsReleased_UpdatesKeyboardInputState()
+        {
+            // Arrange
+            var window = CreateWindow();
+            window.Show();
+
+            // Act
+            this.mockSilkKeyboard.Raise(e => e.KeyUp += It.IsAny<Action<IKeyboard, Key, int>>(),
+                null,
+                Key.Z,
+                0);
+
+            // Assert
+            this.mockKeyInput.VerifyOnce(m => m.SetState(KeyCode.Z, false));
+        }
+
+        [Fact]
+        public void GLWindow_WhenMouseButtonIsPressedDown_UpdatesMouseInputState()
+        {
+            // Arrange
+            var window = CreateWindow();
+            window.Show();
+
+            // Act
+            this.mockSilkMouse.Raise(e => e.MouseDown += It.IsAny<Action<IMouse, SilkMouseButton>>(),
+                null,
+                SilkMouseButton.Left);
+
+            // Assert
+            this.mockMouseInput.VerifyOnce(m => m.SetState(VelaptorMouseButton.LeftButton, true));
+        }
+
+        [Fact]
+        public void GLWindow_WhenMouseButtonIsReleasedDown_UpdatesMouseInputState()
```

Comment:
Refactor unit test name

```suggestion
        public void GLWindow_WhenMouseButtonIsReleased_UpdatesMouseInputState()
```